CoffeeChat/Controller/EditInterestViewController.swift

```diff
@@ -0,0 +1,391 @@
+            if let loc = sourceSection.items.firstIndex(where: { $0.getName() == name }) {
+                targetSection.items.append(sourceSection.items[loc])
+                sourceSection.items.remove(at: loc)
+                sections[sourceIndx] = sourceSection
+                sections[sourceIndx].items.sort(by: { $0.getName() < $1.getName() })
+                sections[targetIndx] = targetSection
+                sections[targetIndx].items.sort(by: { $0.getName() < $1.getName() })
+            }
+        }
+    }
+
+}
+
+// MARK: - UITableViewDelegate
+extension EditInterestViewController: UITableViewDelegate {
+
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let section = sections[indexPath.section]
+        let item = section.items[indexPath.row]
+
+        switch item {
+        case .interest(let interest):
+            let name = interest.name
+            switch section.type {
+            case .yours:
+                moveData(named: name, from: .yours, to: .more)
+            case .more:
+                moveData(named: name, from: .more, to: .yours)
+            }
+
+        case .group(let group):
+            // TODO
+            break
+        }
+
+        tableView.reloadData()
+    }
+}
+
+// MARK: - UITableViewDataSource
+extension EditInterestViewController: UITableViewDataSource {
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return section == 0
+            ? (showingLess && yourCount > hideAfter ? 3 : yourCount)
+            : moreCount
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let section = sections[indexPath.section]
+        let itemType = section.items[indexPath.row]
+
+        switch itemType {
+        case .interest(let interest):
+            guard let cell = tableView.dequeueReusableCell(withIdentifier: OnboardingTableViewCell.reuseIdentifier, for: indexPath) as? OnboardingTableViewCell else { return UITableViewCell() }
+            cell.configure(with: interest)
+            cell.changeColor(isSelected: section.type == .yours)
+            cell.selectionChangesAppearence(false)
+            return cell
+
+        case .group(let group):
+            // TODO
+            return UITableViewCell()
+        }
+    }
+
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        let headerView: UIView
+        if section == 0 && yourCount == 0 {
+            headerView = EditHeaderView(with: "Your Interests", info: "Select at least one interest so we can better help you find a pair!")
+        } else if section == 0 { // Upper Section
+            headerView = EditHeaderView(with: "Your Interests", info: "Tap to deselect")
+        } else { // Lower Section
+            headerView = EditHeaderView(with: "More Interests", info: "Tap to select")
+        }
+        return headerView
```

Comment:
might be better to switch using `sections[section]` so it's more readable for what section it is